<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1241</width>
    <height>642</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>800</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Vipster</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources/vipster.qrc">
    <normaloff>:/images/icon.png</normaloff>:/images/icon.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="0,1,0,0">
    <item>
     <layout class="QHBoxLayout" name="modeLayout">
      <item>
       <widget class="QPushButton" name="camButton">
        <property name="text">
         <string>Camera</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selButton">
        <property name="text">
         <string>Select</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup</string>
        </attribute>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="modButton">
        <property name="text">
         <string>Modify</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <attribute name="buttonGroup">
         <string notr="true">buttonGroup</string>
        </attribute>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="GLWidget" name="openGLWidget">
      <property name="focusPolicy">
       <enum>Qt::ClickFocus</enum>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="displayLayout" stretch="0,0,0,0,0">
      <item>
       <widget class="QLabel" name="camLabel">
        <property name="text">
         <string>Align camera:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="alignLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QPushButton" name="pxButton">
          <property name="text">
           <string>+x</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pyButton">
          <property name="text">
           <string>+y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pzButton">
          <property name="text">
           <string>+z</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="mxButton">
          <property name="text">
           <string>-x</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="myButton">
          <property name="text">
           <string>-y</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="mzButton">
          <property name="text">
           <string>-z</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="displaySpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item alignment="Qt::AlignHCenter">
       <widget class="QLabel" name="multLabel">
        <property name="text">
         <string>Cell multiply:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="multLayout">
        <item>
         <widget class="QSpinBox" name="xMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="yMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="zMultBox">
          <property name="minimum">
           <number>1</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="stepLayout" stretch="0,0,1,0,0,0">
      <item>
       <widget class="QLabel" name="selectLabel">
        <property name="text">
         <string>Select step:</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="controlLayout">
        <item>
         <widget class="QPushButton" name="firstStepButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="preStepButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="playButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="nextStepButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="lastStepButton">
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QSlider" name="stepSlider">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="stepEdit">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="dividerLabel">
        <property name="text">
         <string>/</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="stepLabel">
        <property name="text">
         <string>1</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1241</width>
     <height>30</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuVipster">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menuFrom_existing_Molecule">
     <property name="title">
      <string>&amp;From existing Molecule</string>
     </property>
     <addaction name="actionCopy_single_Step"/>
     <addaction name="actionCopy_Trajector"/>
    </widget>
    <addaction name="actionNew_Molecule"/>
    <addaction name="menuFrom_existing_Molecule"/>
    <addaction name="actionLoad_Molecule"/>
    <addaction name="actionSave_Molecule"/>
    <addaction name="separator"/>
    <addaction name="actionNew_Parameter_Set"/>
    <addaction name="actionSave_Parameter_set"/>
    <addaction name="separator"/>
    <addaction name="actionScreenshot"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionNew_Atom"/>
    <addaction name="actionDelete_Atom_s"/>
    <addaction name="actionCopy_Atom_s"/>
    <addaction name="actionPaste_Atom_s"/>
    <addaction name="separator"/>
    <addaction name="actionHide_Atom_s"/>
    <addaction name="actionShow_Atom_s"/>
    <addaction name="separator"/>
    <addaction name="actionUndo"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout_Vipster"/>
    <addaction name="actionAbout_Qt"/>
   </widget>
   <addaction name="menuVipster"/>
   <addaction name="menuEdit"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="minimumSize">
    <size>
     <width>345</width>
     <height>104</height>
    </size>
   </property>
   <property name="allowedAreas">
    <set>Qt::LeftDockWidgetArea</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QScrollArea" name="scrollArea">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAsNeeded</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="sizeAdjustPolicy">
        <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>341</width>
          <height>566</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <widget class="QListWidget" name="molList"/>
         </item>
         <item>
          <widget class="QPushButton" name="atomTableButton">
           <property name="text">
            <string>Atoms</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTableWidget" name="atomTable">
           <column>
            <property name="text">
             <string>Type</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>x</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>y</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>z</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="cellWidgetButton">
           <property name="text">
            <string>Cell Geometry</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QWidget" name="cellContainer" native="true">
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <property name="leftMargin">
             <number>0</number>
            </property>
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="rightMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <layout class="QHBoxLayout" name="cellDimLayout">
              <item>
               <widget class="QLabel" name="cellVecLabel">
                <property name="text">
                 <string>Cell vectors:</string>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="cellDimSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>40</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <widget class="QLabel" name="cellDimLabel">
                <property name="text">
                 <string>Cell dimension:</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QDoubleSpinBox" name="cellDimBox">
                <property name="minimum">
                 <double>1.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QTableWidget" name="cellVecTable">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>110</height>
               </size>
              </property>
              <row>
               <property name="text">
                <string>x</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>y</string>
               </property>
              </row>
              <row>
               <property name="text">
                <string>z</string>
               </property>
              </row>
              <column>
               <property name="text">
                <string>1</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>2</string>
               </property>
              </column>
              <column>
               <property name="text">
                <string>3</string>
               </property>
              </column>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="cellEditLayout">
           <item>
            <widget class="QCheckBox" name="cellAllBox">
             <property name="text">
              <string>Apply to all Steps</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="cellEditSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QCheckBox" name="cellScaleBox">
             <property name="text">
              <string>Scale coordinates with cell</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="kpointStackButton">
           <property name="text">
            <string>K-Points</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QStackedWidget" name="kpointStack">
           <widget class="QWidget" name="page"/>
           <widget class="QWidget" name="page_2"/>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <action name="actionNew_Molecule">
   <property name="text">
    <string>&amp;New Molecule</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionLoad_Molecule">
   <property name="text">
    <string>&amp;Load Molecule</string>
   </property>
  </action>
  <action name="actionSave_Molecule">
   <property name="text">
    <string>&amp;Save Molecule</string>
   </property>
  </action>
  <action name="actionNew_Parameter_Set">
   <property name="text">
    <string>New &amp;Parameter set</string>
   </property>
  </action>
  <action name="actionSave_Parameter_set">
   <property name="text">
    <string>Sa&amp;ve Parameter set</string>
   </property>
  </action>
  <action name="actionScreenshot">
   <property name="text">
    <string>S&amp;creenshot</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionNew_Atom">
   <property name="text">
    <string>&amp;New Atom</string>
   </property>
   <property name="shortcut">
    <string>N</string>
   </property>
  </action>
  <action name="actionDelete_Atom_s">
   <property name="text">
    <string>&amp;Delete Atom(s)</string>
   </property>
  </action>
  <action name="actionCopy_Atom_s">
   <property name="text">
    <string>&amp;Copy Atom(s)</string>
   </property>
  </action>
  <action name="actionPaste_Atom_s">
   <property name="text">
    <string>&amp;Paste Atom(s)</string>
   </property>
  </action>
  <action name="actionHide_Atom_s">
   <property name="text">
    <string>&amp;Hide Atom(s)</string>
   </property>
  </action>
  <action name="actionShow_Atom_s">
   <property name="text">
    <string>&amp;Show Atom(s)</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="actionAbout_Vipster">
   <property name="text">
    <string>&amp;About Vipster</string>
   </property>
  </action>
  <action name="actionCopy_single_Step">
   <property name="text">
    <string>&amp;Copy single Step</string>
   </property>
  </action>
  <action name="actionCopy_Trajector">
   <property name="text">
    <string>C&amp;opy Trajectory</string>
   </property>
  </action>
  <action name="actionAbout_Qt">
   <property name="text">
    <string>About &amp;Qt</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>glwidget.h</header>
   <slots>
    <slot>setMode(int,bool)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>camButton</tabstop>
  <tabstop>selButton</tabstop>
  <tabstop>modButton</tabstop>
  <tabstop>firstStepButton</tabstop>
  <tabstop>preStepButton</tabstop>
  <tabstop>playButton</tabstop>
  <tabstop>nextStepButton</tabstop>
  <tabstop>lastStepButton</tabstop>
  <tabstop>stepSlider</tabstop>
  <tabstop>pxButton</tabstop>
  <tabstop>pyButton</tabstop>
  <tabstop>pzButton</tabstop>
  <tabstop>mxButton</tabstop>
  <tabstop>myButton</tabstop>
  <tabstop>mzButton</tabstop>
  <tabstop>xMultBox</tabstop>
  <tabstop>yMultBox</tabstop>
  <tabstop>zMultBox</tabstop>
 </tabstops>
 <resources>
  <include location="resources/vipster.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>473</x>
     <y>302</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepEdit</sender>
   <signal>valueChanged(int)</signal>
   <receiver>stepSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1211</x>
     <y>620</y>
    </hint>
    <hint type="destinationlabel">
     <x>1170</x>
     <y>627</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>atomTableButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>atomTable</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>46</x>
     <y>233</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>344</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>kpointStackButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>kpointStack</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>147</x>
     <y>601</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>618</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cellWidgetButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>cellContainer</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>150</x>
     <y>416</y>
    </hint>
    <hint type="destinationlabel">
     <x>160</x>
     <y>511</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout_Vipster</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>about()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>558</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Molecule</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>newMol()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew_Atom</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>editAtoms()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonGroup</sender>
   <signal>buttonToggled(int,bool)</signal>
   <receiver>openGLWidget</receiver>
   <slot>setMode(int,bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>805</x>
     <y>314</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>stepSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setStep(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>619</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>635</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>molList</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>setMol(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>151</y>
    </hint>
    <hint type="destinationlabel">
     <x>353</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>setMol(int)</slot>
  <slot>setStep(int)</slot>
  <slot>editAtoms()</slot>
  <slot>about()</slot>
  <slot>newMol()</slot>
  <slot>setCellVec(int,int)</slot>
  <slot>setCellDim(double)</slot>
 </slots>
 <buttongroups>
  <buttongroup name="buttonGroup"/>
 </buttongroups>
</ui>
